mesonbuild/interpreter/interpreter.py

```diff
@@ -1234,15 +1234,31 @@ def func_add_languages(self, node: mparser.FunctionNode, args: T.Tuple[T.List[st
             return success
 
     @noArgsFlattening
-    @noKwargs
-    def func_message(self, node, args, kwargs):
+    @typed_kwargs(
+        'message',
+        KwargInfo('status', (bool, NoneType), since='0.63.0'),
+        KwargInfo('extra_info', (str, NoneType), since='0.63.0')
+    )
+    def func_message(self, node, args, kwargs: 'kwargs.Message'):
         if len(args) > 1:
             FeatureNew.single_use('message with more than one argument', '0.54.0', self.subproject, location=node)
         args_str = [stringifyUserArguments(i) for i in args]
-        self.message_impl(args_str)
+        self.message_impl(args_str, status=kwargs['status'], extra_info=kwargs['extra_info'])
 
-    def message_impl(self, args):
-        mlog.log(mlog.bold('Message:'), *args)
+    def message_impl(self, args, status: T.Optional[bool] = None, extra_info: T.Optional[str] = None):
+        if status is None:
+            if extra_info is not None:
+                raise InvalidArguments('`extra_info` must be used together with `status`.')
+            mlog.log(mlog.bold('Message:'), *args)
+        else:
+            if len(''.join(args)) == 0:
+                raise InvalidArguments('message must not be empty when `status` kwarg is used.')
+            status_str = mlog.green('YES') if status else mlog.red('NO')
+            args[-1] += ':'
```

Comment:
I don't this it is guaranteed that last arg is a string. It would be weird, but techically they could be int/bool, no? I think you can print the first part of the message with mlog.log(*args, end='') then the status with mlog.log(status_str, extra_info). Would that work?

Reply:
As far as I can see all cases that would ever end up here do `stringifyUserArguments` first, so it seems it would always be a string, no?

Reply:
oh no, forget me, that code is AFTER stringifyUserArguments(). All good.

Reply:
(I agree its not super nice to read but after this PR I plan to refactor some of this anyway in mlog.py itself that hopefully makes this much easier)